syntax = "proto3";

package quiz;

service QuizService {
    rpc Authenticate (AuthenticationRequest) returns (AuthenticationResponse);
    rpc GetQuestion (QuestionRequest) returns (Question);
    rpc SubmitAnswer (AnswerSubmission) returns (AnswerResult);
    rpc GetFinalResult (FinalResultRequest) returns (FinalResult);
}

message AuthenticationRequest {
    string student_id = 1;
    string password = 2;
}

message AuthenticationResponse {
    bool authenticated = 1;
}

message QuestionRequest {
    int32 question_index = 1;
}

message Question {
    int32 question_index = 1;
    string question = 2;
    repeated string options = 3;
}

message AnswerSubmission {
    int32 question_index = 1;
    int32 answer = 2;
}

message AnswerResult {
    bool correct = 1;
}

message FinalResultRequest {
}

message FinalResult {
    int32 total_questions = 1;
    int32 correct_answers = 2;
}
